airflow/providers/microsoft/azure/operators/powerbi.py

```diff
@@ -0,0 +1,200 @@
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+from functools import cached_property
+from typing import TYPE_CHECKING, Sequence
+
+from airflow.configuration import conf
+from airflow.models import BaseOperator, BaseOperatorLink
+from airflow.providers.microsoft.azure.hooks.powerbi import (
+    PowerBIDatasetRefreshException,
+    PowerBIDatasetRefreshFields,
+    PowerBIDatasetRefreshStatus,
+    PowerBIHook,
+)
+
+if TYPE_CHECKING:
+    from airflow.models.taskinstancekey import TaskInstanceKey
+    from airflow.utils.context import Context
+
+
+class PowerBILink(BaseOperatorLink):
+    """Construct a link to monitor a dataset in Power BI."""
+
+    name = "Monitor PowerBI Dataset"
+
+    def get_link(self, operator: BaseOperator, *, ti_key: TaskInstanceKey):
+        url = (
+            f"https://app.powerbi.com"  # type: ignore[attr-defined]
+            f"/groups/{operator.group_id}/datasets/{operator.dataset_id}"  # type: ignore[attr-defined]
+            f"/details?experience=power-bi"
+        )
+
+        return url
+
+
+class PowerBIDatasetRefreshOperator(BaseOperator):
+    """
+    Refreshes a Power BI dataset.
+
+    By default the operator will wait until the refresh has completed before
+    exiting. The refresh status is checked every 60 seconds as a default. This
+    can be changed by specifying a new value for `check_interval`.
+
+    :param dataset_id: The dataset id.
+    :param group_id: The workspace id.
+    :param wait_for_termination: Wait until the pre-existing or current triggered refresh completes before exiting.
+    :param force_refresh: Force refresh if pre-existing refresh found.
+    :param powerbi_conn_id: Airflow Connection ID that contains the connection
+        information for the Power BI account used for authentication.
+    :param timeout: Time in seconds to wait for a dataset to reach a terminal status for non-asynchronous waits. Used only if ``wait_for_termination`` is True.
+    :param check_interval: Number of seconds to wait before rechecking the
+        refresh status.
+    """
+
+    template_fields: Sequence[str] = (
+        "dataset_id",
+        "group_id",
+    )
+    template_fields_renderers = {"parameters": "json"}
+
+    operator_extra_links = (PowerBILink(),)
+
+    def __init__(
+        self,
+        *,  # Indicates all the following parameters must be specified using keyword arguments.
+        dataset_id: str,
+        group_id: str,
+        wait_for_termination: bool = True,
+        force_refresh: bool = False,
+        powerbi_conn_id: str = PowerBIHook.default_conn_name,
+        timeout: int = 60 * 60 * 24 * 7,
+        check_interval: int = 60,
+        deferrable: bool = conf.getboolean("operators", "default_deferrable", fallback=False),
```

Comment:
The MSGraphAsyncOperator is deferrable by default, so I your code is based on that one and the KiotaRequestAdapterHook, you won't need such a boolean, as code will always be async by default, which is better, especially when polling is being done, otherwise workes will be unnecesseraly blocked during polling.

Reply:
Hey @dabla,
My plan is to provide both sync and async modes for the operator because:

Sync Mode:
- Can be used when polling is not needed; the operator runs immediately without blocking workers.
- Suitable for scenarios where users might not have executors set up, which are required for the Airflow operator to run in deferable mode.

Async Mode:
- If polling is required, users can choose to run in deferable mode, leveraging the MSGraphAsyncOperator.